### ISSUE TITLE
TLS 1.1 or TLS 1.2 support

### ISSUE THREAD

**Issue by josemanuelsh** [simpletest/simpletest#34]
Hello,
I am trying to access https://xxx.my.salesforce.com/ using SimpleBrowser, but I get a message from the website to use a browser that supports TLS 1.1 or TLS 1.2 because SourceForce disabled TLS 1.0. There is a public announcement at https://help.salesforce.com/HTViewSolution?id=000221207. 
Is there a way to enable TLS 1.1 or TLS 1.2 using SimpleBrowser?
Thanks!

**Comment by jakoch** [comment]
Hey Jose, 

this is because `https://` requests are done using `tls://` (version 1).
- first `https` is detected, see [http.php#L101](https://github.com/simpletest/simpletest/blob/831a9a2140365a86d0acb4d45f12625af7966895/http.php#L101)
- then, SimpleSecureSocket uses `tls://`, see [socket.php#L324](https://github.com/simpletest/simpletest/blob/master/socket.php#L324)

We need to change this behaviour in SimpleTest and support `tlsv1.1` and `tlsv1.2`,
basically all configured secure transports - as listed by `stream_get_transports()`.

**Workaround**
You could edit the file `socket.php` and replace `tls` with `tlsv1.1` (`tlsv1.2`)
in class `SimpleSecureSocket` function `openSocket()`.

```
public function openSocket($host, $port, &$error_number, &$error, $timeout)
    {
        return parent::openSocket("tlsv11://$host", $port, $error_number, $error, $timeout);
    }
```

Regards, Jens

**Comment by josemanuelsh** [comment]
Great! That worked! Thank you very much!

**Comment by jakoch** [comment]
Glad i could help!

Re-opening the issue, because i have to bugfix SimpleTest first, before we can close :)

**Comment by jakoch** [comment]
Hey!

The bug was fixed in the codebase. I can't remember why i left this open.

Notes: 
- SimpleTest no longer hardcodes tls:// to TLS 1.0 or 1.1. 
- It's a configurable transport now using the $transport variable within SimpleSecureSocket.
- The var defaults to TLS transport 'tlsv1.2'.

Regards, Jens
